src/reactor/mod.rs

```diff
@@ -226,6 +227,18 @@ impl Core {
         res.expect("run should not return until future is done")
     }
 
+    /// Performs one iteration of the event loop, blocking on waiting for events
+    /// for at most `max_wait` (forever if `None`).
+    ///
+    /// It only makes sense to call this method if you've previously spawned
+    /// a future onto this event loop.
+    ///
+    /// `loop { lp.turn(None) }` is equivalent to calling `run` with an
+    /// empty future (one that never finishes).
+    pub fn turn(&mut self, max_wait: Option<Duration>) {
+        self.poll_internal(max_wait, &mut || true).expect("Error in event loop turn");
```

Comment:
Could this call `_run` if we ensured that the loop was turned at least once even if the future is ready?